Homework5/SparseMatrix/SparseMatrix.cs

```diff
@@ -0,0 +1,151 @@
+﻿using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace SparseMatrix
+{
+    public class SparseMatrix : IEnumerable<int>
+    {
+        public int Rows { get; }
+        public int Columns { get; }
+
+        private Dictionary<ElementDirections, int> arrayElements = new Dictionary<ElementDirections, int>();
+
+        public SparseMatrix(int columns, int rows)
+        {
+            if (rows <= 0 || columns <= 0)
+            {
+                throw new ArgumentException();
+            }
+            else
+            {
+                Rows = rows;
+                Columns = columns;
+            }
+
+        }
+
+        public int this[int column, int row]
+        {
+            get
+            {
+                if (row < 0 ||
+                    column < 0 ||
+                    row >= Rows ||
+                    column >= Columns)
+                {
+                    throw new IndexOutOfRangeException();
+                }
+
+                int itemToReturn = default;
+
+                foreach (var element in arrayElements)
```

Comment:
Do not iterate over arrayElements, it is declared as a dictionary. 

Just call`
arrayElements.TryGetValue(new ElementDirections(column, row)), out int vaue) `
or use other dictionary getters/get methods.